test/pack/pack-ze.c

```diff
@@ -17,11 +17,27 @@
 
 #include "level_zero/ze_api.h"
 
-static int ndevices = -1;
 static ze_driver_handle_t global_ze_driver_handle;
 static ze_device_handle_t *global_devices = NULL;
 static ze_context_handle_t ze_context;
 
+int pack_ze_get_ndevices(void)
+{
+    uint32_t driver_count = 0;
+    ret = zeDriverGet(&driver_count, NULL);
+    assert(ret == ZE_RESULT_SUCCESS);
+    assert(driver_count > 0);
+
+    all_drivers = malloc(driver_count * sizeof(ze_driver_handle_t));
+    assert(all_drivers != NULL);
+    ret = zeDriverGet(&driver_count, all_drivers);
+    assert(ret == ZE_RESULT_SUCCESS);
+
+    uint32_t device_count = 0;
+    ret = zeDeviceGet(all_drivers[0], &device_count, NULL);
```

Comment:
code is correct, but why not store the device_count when calling pack_ze_init_devices?
or change pack_ze_init_device to return the count (so one less function to maintain).

Reply:
It seems weird to call `init_devices` when all we need is the number of devices.  I prefer simpler functions that do one thing, rather than large functions that do many things.

Reply:
I meant we call init_device function which returns the device count, so we don't need a separate function to get the count.

Reply:
Then, the function should not do anything more than getting the number of devices.  For example, it cannot create any contexts like the current code does.

Btw, I just realized that the ZE version of the function is completely wrong.  :-).  It's not returning anything.  :-). I'll fix it in a bit, but I might need you to test it out.

Reply:
Pushed my changes.  FYI, I don't want a function that does "init_devices_and_get_ndevices".  I prefer having two simpler functions instead: "init_devices" and "get_ndevices".  This allows the caller to call get_ndevices any number of times without having to worry about it.  On the other hand, "init_devices" clearly goes together with "finalize_devices" to create/free resources.